projects/plugins/jetpack/modules/masterbar/profile-edit/class-wpcom-user-profile-fields-revert.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Manage User profile fields.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Responsible with preventing the back-end default implementation to save the fields that are managed on WP.com profiles.
+ *
+ * Class Profile_Edit_Filter_Fields
+ */
+class WPCOM_User_Profile_Fields_Revert {
+
+	/**
+	 * Jetpack connection manager object.
+	 *
+	 * @var Connection_Manager
+	 */
+	private $connection_manager;
+
+	/**
+	 * Profile_Edit_Filter_Fields constructor.
+	 *
+	 * @param string             $current_page The current page slug.
+	 * @param Connection_Manager $connection_manager The connection manager.
+	 */
+	public function __construct( $current_page, Connection_Manager $connection_manager ) {
+		// We'll apply this filter only on profile edit page to be sure we limit the impact of the change.
+		if ( 'user-edit.php' !== $current_page && 'profile.php' !== $current_page ) {
+			return;
+		}
+
+		$this->connection_manager = $connection_manager;
+
+		\add_filter( 'wp_pre_insert_user_data', array( $this, 'revert_display_name_on_wp_admin_profile_update' ), 10, 3 );
+		\add_filter( 'insert_user_meta', array( $this, 'revert_user_meta_on_wp_admin_profile_change' ), 10, 3 );
+
+		/**
+		 * Disable notification on E-mail changes for Atomic WP-Admin Edit Profile. (for WP.com we use a different section for changing the E-mail).
+		 *
+		 * We need this because WP.org uses a custom flow for E-mail changes.
+		 */
+		\remove_action( 'personal_options_update', 'send_confirmation_on_profile_email' );
+	}
+
+	/**
+	 * Filter the built-in user profile fields.
+	 *
+	 * @param array    $data            {
+	 *                                  Values and keys for the user.
+	 *
+	 * @type string    $user_login      The user's login. Only included if $update == false
+	 * @type string    $user_pass       The user's password.
+	 * @type string    $user_email      The user's email.
+	 * @type string    $user_url        The user's url.
+	 * @type string    $user_nicename   The user's nice name. Defaults to a URL-safe version of user's login
+	 * @type string    $display_name    The user's display name.
+	 * @type string    $user_registered MySQL timestamp describing the moment when the user registered. Defaults to
+	 *                                   the current UTC timestamp.
+	 * }
+	 *
+	 * @param bool     $update          Whether the user is being updated rather than created.
+	 * @param int|null $id              ID of the user to be updated, or NULL if the user is being created.
+	 *
+	 * @return array
+	 */
+	public function revert_display_name_on_wp_admin_profile_update( $data, $update, $id ) {
```

Comment:
We should rename this now to something more generic.
```suggestion
	public function revert_user_data_on_wp_admin_profile_update( $data, $update, $id ) {
```